### PR TITLE
Fix the connection error

### PR DESCRIPTION
**Pull request by seii-saintway**
1. public_api.py

Traceback (most recent call last):
  File "/opt/tljh/user/lib/python3.6/site-packages/urllib3/connection.py", line 159, in _new_conn
  File "/opt/tljh/user/lib/python3.6/site-packages/urllib3/util/connection.py", line 57, in create_connection
  File "/opt/tljh/user/lib/python3.6/socket.py", line 745, in getaddrinfo
OSError: [Errno 23] Too many open files in system
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/opt/tljh/user/lib/python3.6/site-packages/urllib3/connectionpool.py", line 600, in urlopen
  File "/opt/tljh/user/lib/python3.6/site-packages/urllib3/connectionpool.py", line 343, in _make_request
  File "/opt/tljh/user/lib/python3.6/site-packages/urllib3/connectionpool.py", line 839, in _validate_conn
  File "/opt/tljh/user/lib/python3.6/site-packages/urllib3/connection.py", line 301, in connect
  File "/opt/tljh/user/lib/python3.6/site-packages/urllib3/connection.py", line 168, in _new_conn
urllib3.exceptions.NewConnectionError: <urllib3.connection.VerifiedHTTPSConnection object at 0x7f8bf407e940>: Failed to establish a new connection: [Errno 23] Too many open files in system
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/opt/tljh/user/lib/python3.6/site-packages/requests/adapters.py", line 449, in send
  File "/opt/tljh/user/lib/python3.6/site-packages/urllib3/connectionpool.py", line 638, in urlopen
  File "/opt/tljh/user/lib/python3.6/site-packages/urllib3/util/retry.py", line 399, in increment
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='public.bitbank.cc', port=443): Max retries exceeded with url: /eth_jpy/ticker (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f8bf407e940>: Failed to establish a new connection: [Errno 23] Too many open files in system',))
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/opt/tljh/user/lib/python3.6/site-packages/python_bitbankcc/public_api.py", line 46, in get_ticker
  File "/opt/tljh/user/lib/python3.6/site-packages/python_bitbankcc/public_api.py", line 41, in _query
  File "/opt/tljh/user/lib/python3.6/site-packages/requests/api.py", line 75, in get
  File "/opt/tljh/user/lib/python3.6/site-packages/requests/api.py", line 60, in request
  File "/opt/tljh/user/lib/python3.6/site-packages/requests/sessions.py", line 533, in request
  File "/opt/tljh/user/lib/python3.6/site-packages/requests/sessions.py", line 646, in send
  File "/opt/tljh/user/lib/python3.6/site-packages/requests/adapters.py", line 516, in send
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='public.bitbank.cc', port=443): Max retries exceeded with url: /eth_jpy/ticker (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f8bf407e940>: Failed to establish a new connection: [Errno 23] Too many open files in system',))

2. private_api.py

Traceback (most recent call last):
  File "/opt/tljh/user/lib/python3.6/site-packages/python_bitbankcc/private_api.py", line 79, in get_asset
    return self._get_query('/user/assets', {})
  File "/opt/tljh/user/lib/python3.6/site-packages/python_bitbankcc/private_api.py", line 67, in _get_query
    response = requests.get(uri, headers=headers)
  File "/opt/tljh/user/lib/python3.6/site-packages/requests/api.py", line 75, in get
    return request('get', url, params=params, **kwargs)
  File "/opt/tljh/user/lib/python3.6/site-packages/requests/api.py", line 60, in request
    return session.request(method=method, url=url, **kwargs)
  File "/opt/tljh/user/lib/python3.6/site-packages/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/opt/tljh/user/lib/python3.6/site-packages/requests/sessions.py", line 646, in send
    r = adapter.send(request, **kwargs)
  File "/opt/tljh/user/lib/python3.6/site-packages/requests/adapters.py", line 498, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', OSError("(32, 'EPIPE')",))

### CODE REVIEW

File: python_bitbankcc/private_api.py [hunk 1]
```diff
@@ -64,16 +64,18 @@ def _get_query(self, path, query):
         logger.debug('GET: ' + data)
         headers = make_header(data, self.api_key, self.api_secret)
         uri = self.end_point + path + urlencode(query)
-        response = requests.get(uri, headers=headers)
-        return error_parser(response.json())
+        import contextlib
```
Comment (junderw) [inline python_bitbankcc/private_api.py]: imports should be at the top.

File: python_bitbankcc/private_api.py [hunk 2]
```diff
@@ -64,16 +64,18 @@ def _get_query(self, path, query):
...
+            return error_parser(response.json())
     
     def _post_query(self, path, query):
         data = json.dumps(query)
         logger.debug('POST: ' + data)
         headers = make_header(data, self.api_key, self.api_secret)
         uri = self.end_point + path
-        response = requests.post(uri, data=data, headers=headers)
-        return error_parser(response.json())
+        import contextlib
```
Comment (junderw) [inline python_bitbankcc/private_api.py]: imports should be at the top.

File: python_bitbankcc/public_api.py [hunk 3]
```diff
@@ -38,8 +38,9 @@ def __init__(self):
         self.end_point = 'https://public.bitbank.cc'
     
     def _query(self, query_url):
-        response = requests.get(query_url)
-        return error_parser(response.json())
+        import contextlib
```
Comment (junderw) [inline python_bitbankcc/public_api.py]: imports should be at the top.

**Review by junderw — CHANGES_REQUESTED** [review]
LGTM besides changes

**Comment by seii-saintway** [comment]
> LGTM besides changes

I've moved the imports.

**Review by junderw — APPROVED** [review]

**Comment by junderw** [comment]
Thanks!

**Pull request merged**